dan_layer/storage_sqlite/migrations/2022-10-03-178235_create_shard_store/up.sql

```diff
@@ -90,21 +90,19 @@ create table leader_proposals (
 
 create unique index leader_proposals_index on leader_proposals (payload_id, shard_id, payload_height, node_hash);
 
-create table objects (
-    id integer not null primary key AUTOINCREMENT,
-    shard_id blob not NULL,
-    payload_id blob not NULL,
-    object_id blob not NULL,
-    node_height bigint not NULL,
-    current_state text not NULL,
-    object_pledge text not NULL -- TODO: can it be non null ?
-);
-
 
-create table substate_changes (
+create table substates (
     id integer not null primary key AUTOINCREMENT,
+    substate_type text not null,
     shard_id blob not NULL,
-    substate_change text not null,
-    tree_node_hash blob not NULL
+    node_height bigint not NULL,
+    data blob null,
+    created_by_payload_id blob not null,
```

Comment:
Maybe I am misunderstanding it. But, can't we have access to these fields by deserializing `substate_type` into a `SubstateState` type ?